source/MRMesh/MRColorMapAggregator.h

```diff
@@ -0,0 +1,92 @@
+#pragma once
+#include "MRMeshFwd.h"
+#include "MRVector.h"
+#include "MRColor.h"
+#include "MRId.h"
+#include "MRBitSet.h"
+
+
+namespace MR
+{
+/**
+ * @brief Class for aggregate several color map in one
+ * @detail Color maps are aggregated according order
+ */
+template<typename Tag>
+class ColorMapAggregator
+{
+public:
+    using ColorMap = Vector<Color, Id<Tag>>;
+    using ElementBitSet = TaggedBitSet<Tag>;
+
+    ColorMapAggregator() = default;
+
+    /// set default (background) color
+    MRMESH_API void setDefaultColor( const Color& color );
+
+    /**
+     * @brief add color map after all (more priority)
+     * @param colorMap color map for elements
+     * @param elementBitSet bitset of elements for which the color map is applied
+     */
+    MRMESH_API void pushBack( const ColorMap& colorMap, const ElementBitSet& elementBitSet );
+
+    /**
+     * @brief insert color map after element #i (0 - minimum priority)
+     * @param colorMap color map for elements
+     * @param elementBitSet bitset of elements for which the color map is applied
+     */
+    MRMESH_API void insert( int i, const ColorMap& colorMap, const ElementBitSet& elementBitSet );
+
+    /**
+     * @brief replace color map in #i position
+     * @param colorMap color map for elements
+     * @param elementBitSet bitset of elements for which the color map is applied
+     */
+    MRMESH_API void replace( int i, const ColorMap& colorMap, const ElementBitSet& elementBitSet );
+
+    /// reset all accumulated color map
+    MRMESH_API void reset();
+
+    /// get number of accumulated color maps
+    size_t getColorMapNumber() { return dataSet_.size(); };
+
+    /// get color map by index
+    const ColorMap& getColorMap( int i ) { return dataSet_[i].colorMap; };
```

Comment:
i think this should return `Data` object also, mb better make it public?

Reply:
fixed